libc/src/stdio/scanf_core/vfscanf_internal.h

```diff
@@ -12,17 +12,41 @@
 #include "src/__support/File/file.h"
 #include "src/__support/arg_list.h"
 #include "src/__support/macros/config.h"
+#include "src/__support/macros/properties/architectures.h"
 #include "src/stdio/scanf_core/reader.h"
 #include "src/stdio/scanf_core/scanf_main.h"
 
+#if defined(LIBC_TARGET_ARCH_IS_GPU)
+#include "src/stdio/ferror.h"
+#include "src/stdio/getc.h"
+#include "src/stdio/ungetc.h"
+#endif
+
 #include "hdr/types/FILE.h"
 #include <stddef.h>
 
 namespace LIBC_NAMESPACE_DECL {
 
 namespace internal {
 
-#ifndef LIBC_COPT_STDIO_USE_SYSTEM_FILE
+#if defined(LIBC_TARGET_ARCH_IS_GPU)
+// Since ungetc_unlocked isn't always available, we don't acquire the lock for
+// system files.
+LIBC_INLINE void flockfile(::FILE *) { return; }
+
+LIBC_INLINE void funlockfile(::FILE *) { return; }
+
+LIBC_INLINE int getc(void *f) {
+  return LIBC_NAMESPACE::getc(reinterpret_cast<::FILE *>(f));
```

Comment:
If you're intentionally using the public entrypoints here, you need to add a specific comment explaining why it's important in this case, since we generally don't allow that.

Reply:
Figured it's alright since it's the namespace qualified one, but not doing this would require a bit of a mess.

Reply:
the namespace isn't the issue, it's the fact that it makes entrypoints less independent. If entrypoints include each other directly then you have to include them as a group, you can't just get scanf without ungetc. This causes problems when trying to rollout individual functions.

Reply:
Thought that's what dependencies were for, but I did add a comment about it.

Reply:
The dependencies mean the build won't break, but that doesn't help with the rollout problem.

Reply:
I'm okay with leaving it as just a comment, the thing I want is to make it clear that this is unusual and people shouldn't copy it. I'm willing to let the GPU functions be less independent since there's no existing libc to overlay, but I want to make sure the functions on other targets stay independent.

Reply:
Added a bit extra